bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/data/v2/wrappers/Filters.java

```diff
@@ -0,0 +1,499 @@
+     * be present in a binary key.
+     */
+    public Filter regex(String regex) {
+      return new SimpleFilter(
+          RowFilter.newBuilder().setRowKeyRegexFilter(ByteString.copyFromUtf8(regex)).build());
+    }
+
+    /**
+     * Matches only cells from rows whose keys equal the value. In other words, passes through the
+     * entire row when the key matches, and otherwise produces an empty row.
+     */
+    public Filter exactMatch(ByteString value) {
+      return new SimpleFilter(
+          RowFilter.newBuilder().setRowKeyRegexFilter(RegexUtil.literalRegex(value)).build());
+    }
+
+    /**
+     * Matches all cells from a row with `probability`, and matches no cells from the row with
+     * probability 1-`probability`.
+     */
+    public Filter sample(double probability) {
+      return new SimpleFilter(RowFilter.newBuilder().setRowSampleFilter(probability).build());
+    }
+  }
+
+  public static class FamilyFilter {
+
+    /**
+     * Matches only cells from columns whose families satisfy the given <a
+     * href="https://github.com/google/re2/wiki/Syntax>RE2 regex</a>. For technical reasons, the
+     * regex must not contain the `:` character, even if it is not being used as a literal. Note
+     * that, since column families cannot contain the new line character `\n`, it is sufficient to
+     * use `.` as a full wildcard when matching column family names.
+     */
+    public Filter regex(String regex) {
+      return new SimpleFilter(RowFilter.newBuilder().setFamilyNameRegexFilter(regex).build());
+    }
+
+    /** Matches only cells from columns whose families match the value. */
+    public Filter exactMatch(String value) {
+      return new SimpleFilter(
+          RowFilter.newBuilder().setFamilyNameRegexFilter(RegexUtil.literalRegex(value)).build());
+    }
+  }
+
+  public static class QualifierFilter {
+
+    /**
+     * Matches only cells from columns whose qualifiers satisfy the given <a
+     * href="https://github.com/google/re2/wiki/Syntax>RE2 regex</a>. Note that, since column
+     * qualifiers can contain arbitrary bytes, the `\C` escape sequence must be used if a true
+     * wildcard is desired. The `.` character will not match the new line character `\n`, which may
+     * be present in a binary qualifier.
+     */
+    public Filter regex(String regex) {
+      return new SimpleFilter(
+          RowFilter.newBuilder()
+              .setColumnQualifierRegexFilter(ByteString.copyFromUtf8(regex))
+              .build());
+    }
+
+    /** Matches only cells from columns whose qualifiers equal the value. */
+    public Filter exactMatch(ByteString value) {
+      return new SimpleFilter(
+          RowFilter.newBuilder()
+              .setColumnQualifierRegexFilter(RegexUtil.literalRegex(value))
+              .build());
+    }
+
+    /**
+     * Matches only cells from columns within the given range.
+     */
+    public static class RangeBuilder {
```

Comment:
I think it would be nicer to:
- pull this up one level
- call it QualifierRangeFilter
- make it extend Filter, replacing build() with toProto()
- while keeping the empty range() here, add an overload for range(closedStart, openEnd)